### PR TITLE
Handles HTTP requests/responses

### PR DESCRIPTION
**Pull request by seoulection**

### CODE REVIEW

File: src/main/java/webserver/HttpHandler.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        this.clientSocket = clientSocket;
+    }
+
+    public void run() {
+        int statusCode = 200;
+        String contentType = "text/html";
+        String content;
+        try {
+            PrintWriter out = new PrintWriter(clientSocket.getOutputStream(), true);
+            BufferedReader br = SocketIO.createSocketReader(clientSocket);
```
Comment (klkelley) [inline src/main/java/webserver/HttpHandler.java]: Can this variable be renamed to be a bit more descriptive?

File: src/main/java/webserver/HttpHandler.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.net.Socket;
+
+public class HttpHandler implements Runnable {
+    Socket clientSocket;
+
+    public HttpHandler(Socket clientSocket) {
+        this.clientSocket = clientSocket;
+    }
+
+    public void run() {
```
Comment (klkelley) [inline src/main/java/webserver/HttpHandler.java]: This function does appears to do a couple different things right now, I think once you have to support more functionality you'll begin to feel some pain points here and need to refactor.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Java Todo List
-[![Build Status](https://travis-ci.org/seoulection/webserver.svg?branch=sockets)](https://travis-ci.org/seoulection/webserver)
+[![Build Status](https://travis-ci.org/seoulection/webserver.svg?branch=req-res)](https://travis-ci.org/seoulection/webserver)
```
[the remark this answers is not in the document]
Comment (seoulection) [reply README.md]: Ah, good to know! 😅

**Comment by seoulection** [comment]
> Nice job, this is a great start!
> 
> I left a few comments, they aren't blockers and I'd be fine if you wanted to submit those in a followup refactor PR for any of those suggestions.

Thanks for your feedback Kara! I'll go ahead and make the changes to `HttpResponse` and `HttpResponseTest` to use the builder pattern, as well as changing the `BufferedReader` variable to a more descriptive name, and push those changes to this PR. I'll address all other refactoring (in this PR and the walking skeleton PR) in a separate `refactor` branch!

File: src/main/java/webserver/HttpHandler.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            BufferedReader input = SocketIO.createSocketReader(clientSocket);
+            HttpRequest httpRequest = new HttpRequest(SocketIO.readFromInputStream(input));
+            String path = httpRequest.getPath();
+            if (path.equals("/")) {
+                content = "<h1>Hello World!</h1>";
+            } else {
+                statusCode = 404;
+                content = "<h1>Page not found</h1>";
+            }
+            output.println(HttpResponse.response(statusCode, contentType, content));
```
Comment (omarnyte) [inline src/main/java/webserver/HttpHandler.java]: It seems a little strange to me that a class called `HttpResponse` has a method named `response`. Without looking at the implementation, I'd be a little unsure about what that method is doing. Is there a name that you can give that method in order to remove some of that ambiguity?

File: src/main/java/webserver/HttpResponseBuilder.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package webserver;
+
+public class HttpResponseBuilder {
```
Comment (omarnyte) [inline src/main/java/webserver/HttpResponseBuilder.java]: Good work bringing in the builder pattern! Your responses are only going to get bigger with new requirements. It's good seeing you implement this pattern know that you would have had to pass in three parameters to construct a response.

**Pull request merged**